metrics/gaugeReporter.go

```diff
@@ -0,0 +1,26 @@
+package metrics
+
+import "time"
+
```

Comment:
Just thinking, could we move this refactoring out from this release and leave it in a separate PR? The reason: Since we are trying to release what's battle tested, we'd avoid any code changes for this release?
wdyt?

Reply:
Okay. Taking the feedbacks on another PR.